src/pedal/mod.rs

```diff
@@ -1,15 +1,18 @@
+mod delay;
 mod transparent;
 
+pub use delay::Delay;
 pub use transparent::Transparent;
 
-use std::fmt::{Debug, Display};
+use crate::Result;
+use std::fmt::Display;
 
 pub type Boxed = Box<dyn Pedal>;
 
-pub trait Pedal: Debug + Send + Sync {
+pub trait Pedal: Send + Sync {
     fn name(&self) -> String;
 
-    fn process(&mut self, input: &[f32], output: &mut [f32]);
+    fn process(&mut self, input: &[f32], output: &mut [f32]) -> Result<()>;
```

Comment:
Some of the ring buffer function can fail, so I changed this to return a `Result`.